AllJoyn/Platform/BridgeHost/App.xaml.cs

```diff
@@ -0,0 +1,359 @@
+            }
+        }
+
+        ApplicationDataCompositeValue ValueSetToAppDataCompositeValue(ValueSet valuset)
+        {
+            ApplicationDataCompositeValue appCompositeData = new ApplicationDataCompositeValue();
+            foreach (var key in valuset.Keys)
+            {
+                appCompositeData.Add(key, valuset[key]);
+            }
+            return appCompositeData;
+        }
+
+        ValueSet AppDataCompositeValueToValueSet(ApplicationDataCompositeValue appCompositeData)
+        {
+            ValueSet values = new ValueSet();
+            foreach (var key in appCompositeData.Keys)
+            {
+                values.Add(key, appCompositeData[key]);
+            }
+            return values;
+        }
+
+        async Task AddDeviceToAllJoynAsync(ValueSet onboardingData)
+        {
+            // Check the cached onboarding devices, whether the specified device is already onboarded.
+            var appData = Windows.Storage.ApplicationData.Current.LocalSettings;
+            var onboardedDevicesContainer = appData.CreateContainer(OnboardedDevicesContainerName, ApplicationDataCreateDisposition.Always);
+            
+            if (!onboardedDevicesContainer.Values.ContainsKey(GetOnboardingId(onboardingData)))
+            {
+                /// Populate DeviceInfo object
+                var device = new BridgeRT.DeviceInfo();
+                device.Name = onboardingData["name"] as string;
+                DataContractJsonSerializer jsonSerializer = new DataContractJsonSerializer(typeof(DeviceProps));
+                MemoryStream memStream = new MemoryStream();
+                jsonSerializer.WriteObject(memStream, DeviceProps.CreateInstance(onboardingData["id"] as string, onboardingData["access_token"] as string));
+                memStream.Position = 0;
+                StreamReader reader = new StreamReader(memStream);
+                device.Props = reader.ReadToEnd();
+
+                var translatorJsSharingToken = onboardingData["translatorJs"] as string;
+                var schemaSharingToken = onboardingData["schema"] as string;
+
+                var schemaFile = await SharedStorageAccessManager.RedeemTokenForFileAsync(schemaSharingToken);
+                var schema = await FileIO.ReadTextAsync(schemaFile);
+                var translatorJsFile = await SharedStorageAccessManager.RedeemTokenForFileAsync(translatorJsSharingToken);
+                var translatorJs = await FileIO.ReadTextAsync(translatorJsFile);
+
+                var bridge = new BridgeRT.BridgeJs();
+
+                await bridge.AddDeviceAsync(device, schema, translatorJs, ".");
+                StringBuilder responseBuilder = new StringBuilder();
+
+                // Cache the onboarding data and add the device to the bridge
+                var translatorsContainer = appData.CreateContainer(onboardingData["category"] as string, ApplicationDataCreateDisposition.Always);
```

Comment:
This is client controlled, as documented in readme.md. It is device type. - WinkThermostat, WinkLightbulb etc...

---
In reply to: [75187182](https://github.com/tjaffri/msiot-samples/pull/9#discussion_r75187182) [](ancestors = 75187182)